compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -284,6 +285,25 @@ impl<'tcx> SizeSkeleton<'tcx> {
                     ),
                 }
             }
+            ty::Array(ty, len) if len.ty() == tcx.types.usize => {
+                match SizeSkeleton::compute(ty, tcx, param_env)? {
+                    SizeSkeleton::Known(s) => {
+                        if let Some(c) = len.try_eval_usize(tcx, param_env) {
+                            Ok(SizeSkeleton::Known(c * s))
+                        } else {
+                            let len = tcx.expand_abstract_consts(len);
+                            let prev = ty::Const::from_usize(tcx, s.bytes());
+                            Ok(SizeSkeleton::Generic(combine_sorted_expr(
+                                tcx, param_env, len, prev,
+                            )))
+                        }
+                    }
+                    SizeSkeleton::Pointer { .. } => Err(err),
+                    SizeSkeleton::Generic(g) => {
+                        Ok(SizeSkeleton::Generic(combine_sorted_expr(tcx, param_env, len, g)))
```

Comment:
How come this doesn't call `tcx.expand_abstract_consts(len)`?

Reply:
oops yes definitely should here